neo/public/javascripts/map/controllers.js

```diff
@@ -371,19 +373,31 @@ angular.module('cloudberry.map', ['leaflet-directive', 'cloudberry.common'])
     function drawMap(result) {
 
       var colors = $scope.styles.colors;
+      var sentimentColors = $scope.styles.sentimentColors;
 
       function getColor(d) {
-        if(!d || d <= 0) {
-          d = 0;
-        } else if (d ===1 ){
-          d = 1;
-        } else {
-          d = Math.ceil(Math.log10(d));
-          if(d <= 0) // treat smaller counts the same as 0
-            d = 0
+        if($scope.doSentiment){  // 0 <= d <= 4
```

Comment:
could you make two functions `getSentimentColor` and `getCountColor`, then make this `getColor` more simpler?